src/openrct2-ui/windows/Changelog.cpp

```diff
@@ -41,273 +41,293 @@ static constexpr const rct_string_id WINDOW_TITLE = STR_CHANGELOG_TITLE;
-    return env->GetFilePath(PATHID::CHANGELOG);
-}
-
-static std::string GetChangelogText()
-{
-    auto path = GetChangelogPath();
-#if defined(_WIN32) && !defined(__MINGW32__)
-    auto pathW = String::ToWideChar(path);
-    auto fs = std::ifstream(pathW, std::ios::in);
-#else
-    auto fs = std::ifstream(path, std::ios::in);
-#endif
-    if (!fs.is_open())
+    /**
+     * @brief Get the absolute path for the changelog file
+     *
+     * @return std::string
+     */
+    std::string GetChangelogPath()
     {
-        throw std::runtime_error("Unable to open " + path);
+        auto env = GetContext()->GetPlatformEnvironment();
+        return env->GetFilePath(PATHID::CHANGELOG);
     }
-    return std::string((std::istreambuf_iterator<char>(fs)), std::istreambuf_iterator<char>());
-}
 
-static bool window_changelog_read_file()
-{
-    std::string _changelogText;
-    try
+    /**
+     * @brief Attempts to read the changelog file, returns true on success
+     *
+     */
+    bool ReadChangelogFile()
     {
-        _changelogText = GetChangelogText();
+        std::string _changelogText;
+        try
+        {
+            _changelogText = GetChangelogText();
+        }
+        catch (const std::bad_alloc&)
+        {
+            log_error("Unable to allocate memory for changelog.txt");
+            return false;
+        }
+        catch (const std::exception&)
+        {
+            log_error("Unable to read changelog.txt");
+            return false;
+        }
+
+        ProcessChangelogText(_changelogText);
+        return true;
     }
-    catch (const std::bad_alloc&)
+
+    /**
+     * @brief Ingests a string of text and splits it into lines for the changelog and updates the longest line width for
+     * scrolling purposes
+     *
+     * @param text
+     */
+    void ProcessChangelogText(const std::string& text)
     {
-        log_error("Unable to allocate memory for changelog.txt");
-        return false;
+        std::string::size_type pos = 0;
+        std::string::size_type prev = 0;
+        while ((pos = text.find("\n", prev)) != std::string::npos)
+        {
+            _changelogLines.push_back(text.substr(prev, pos - prev));
+            prev = pos + 1;
+        }
+
+        // To get the last substring (or only, if delimiter is not found)
+        _changelogLines.push_back(text.substr(prev));
+
+        _changelogLongestLineWidth = 0;
+        for (const auto& line : _changelogLines)
+        {
+            int32_t linewidth = gfx_get_string_width(line.c_str(), FontSpriteBase::MEDIUM);
+            _changelogLongestLineWidth = std::max(linewidth, _changelogLongestLineWidth);
+        }
     }
-    catch (const std::exception&)
+};
+
+rct_window* window_changelog_open(int personality)
+{
+    auto* window = window_bring_to_front_by_class(WC_CHANGELOG);
+    if (window == nullptr)
     {
-        log_error("Unable to read changelog.txt");
-        return false;
+        // Create a new centered window
```

Comment:
```suggestion
        // Create a new centred window
```

Very minor nitpick, but we used British English in the codebase.